[UX-53] rpk: add well-known protobuf types to rpk

r-vasquez opened the pull request:
This is the same set of well-known types that both Redpanda and Redpanda Console already support.

This allows users to use the whole suite and encode/decode using these well-known types.

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

### Improvements

* rpk now supports well-known protobuf types when encoding/decoding records using Schema Registry.

r-vasquez committed:
rpk: add well-known protobuf types to rpk

This is the same set of well-known types that
both Redpanda and Redpanda Console already support

This allows users to use the whole suite and
encode/decode using these well-known types.

One rpk unit tests verify that the embedded types
are the same as the ones in the Redpanda repo,
hence the need to trigger rpk tests on changes
in pandaproxy/schema_registry/protobuf

vbotbuildovich commented:
<!--RETRIED FAILURES-->
<!--5c9ed3bf-9009-4757-9037-bb5f07a32d4b-->
<!--RETRIED FAILURES-->
<!--d004a99b-1708-484d-9d13-2446b91c6620-->
<!--RETRIED FAILURES-->
<!--ba2e691d-804b-48ef-95cb-2f45cb4b1899-->
<!--RETRIED FAILURES-->
<!--ac97ba78-fc28-4e0a-b6aa-b93c00055b47-->
<!--body-->
ducktape was retried in https://buildkite.com/redpanda/redpanda/builds/59421#01939e90-a445-48ca-a8a7-f1fb2437c7bd
ducktape was retried in https://buildkite.com/redpanda/redpanda/builds/59564#0193b278-c7c1-435a-89fe-a3db40745a4e
ducktape was retried in https://buildkite.com/redpanda/redpanda/builds/59564#0193b278-c7c0-4c97-8a4a-3e00cf241c30
ducktape was retried in https://buildkite.com/redpanda/redpanda/builds/59624#0193b85a-cf42-4aae-a646-986e0e7f9fe9
<!--/body-->

rockwotj commented on src/go/rpk/pkg/serde/embed/BUILD:
```diff
@@ -0,0 +1,38 @@
+    embedsrcs = [
+        "protobuf/confluent/meta.proto",
+        "protobuf/confluent/types/decimal.proto",
+        "protobuf/google/type/README.md",
```
does this need to be embedded?

r-vasquez replied:
Good call, I've changed the Go directive to only include `.proto` files. :+1:

rockwotj commented on src/go/rpk/pkg/serde/embed/BUILD:
```diff
@@ -0,0 +1,38 @@
+        "protobuf/google/type/postal_address.proto",
+        "protobuf/google/type/quaternion.proto",
+        "protobuf/google/type/timeofday.proto",
+        "protobuf/.clang-format",
```
Is this needed?

r-vasquez replied:
Same as above, I removed it from the embedded files. Thanks

rockwotj commented on src/go/rpk/pkg/serde/embed/embed_test.go:
```diff
@@ -0,0 +1,44 @@
+	"github.com/stretchr/testify/require"
+)
+
+func TestEmbeddedFiles(t *testing.T) {
+	t.Run("Test Embedded files in rpk, equal to Redpanda", func(t *testing.T) {
+		// /src/v/pandaproxy/schema_registry/protobuf
+		redpandaProtoFS := os.DirFS("../../../../../v/pandaproxy/schema_registry/protobuf/")
+		redpandaMap := make(map[string]string)
+		err := fs.WalkDir(redpandaProtoFS, ".", func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+			if d.IsDir() {
+				return nil
+			}
+			if filepath.Ext(path) == ".proto" {
+				data, err := fs.ReadFile(redpandaProtoFS, path)
+				if err == nil {
+					redpandaMap[path] = string(data)
+				}
+			}
+			return nil
+		})
+
+		embeddedMap, err := CommonProtoFileMap()
+		require.NoError(t, err)
+
+		for path, embedContent := range embeddedMap {
+			if rpContent, ok := redpandaMap[path]; ok {
+				require.Equalf(t, rpContent, embedContent, "Contents of %v have changed vs the embedded rpk files", path)
+			} else {
+				t.Fatalf("%s not found in Redpanda files", path)
+			}
+		}
+	})
+}
```
Will this test run if only broker code changes?

rockwotj reviewed (dismissed):
LGTM just one question about keeping the broker/rpk in sync

rockwotj commented on .github/workflows/rpk-build.yml:
```diff
@@ -13,10 +13,12 @@ on:
   pull_request:
     paths:
       - 'src/go/rpk/**'
+      - 'src/v/pandaproxy/schema_registry/protobuf'
```
Does this need `/**` at the end?

r-vasquez replied:
Yes, you are right. Added in https://github.com/redpanda-data/redpanda/compare/f1de4de3714eb845c559fd97cede144b488de8d7..00ea43184a0ad352d0d761f3c87d7b434b4f8695

r-vasquez commented:
Last force pushes include:
- Rebasing with Dev
- Fixing a dependencies issue with Bazel, in which I have to manually specify the need for the `embed/` package to get the files from `pandaproxy/schema_registry/protobuf`
- Changing the rules of the rpk's go test, to be triggered when the `schema_registry/protobuf` package changes.

bojand commented on src/go/rpk/pkg/serde/embed/embed.go:
```diff
@@ -0,0 +1,64 @@
+			if err != nil {
+				return err
+			}
+			if d.IsDir() {
```
Minor subjective change... I think the 2 checks could be combined into one that may be more readable?
```go
if d.IsDir() || filepath.Ext(path) != ".proto" {
  return nil
}

data, err := fs.ReadFile(protoFS, path)
if err == nil {
  protoMap[path] = string(data)
}
```

r-vasquez replied:
:+1: Agree, added the change in https://github.com/redpanda-data/redpanda/compare/00ea43184a0ad352d0d761f3c87d7b434b4f8695..0404d15560b25c7026b593382887829c77a03ceb

vbotbuildovich commented:
<!--UNIT FAILURES-->
<!--285ce670-7694-45de-8367-1c641f01ac51-->
<!--body-->
the below tests from https://buildkite.com/redpanda/redpanda/builds/59507#0193ad48-23ca-4b70-afa2-f3557643c49f have failed and will be retried
```
gtest_raft_rpunit
```
<!--/body-->

bojand commented on src/go/rpk/pkg/serde/embed/embed_test.go:
```diff
@@ -0,0 +1,44 @@
+			}
+			if d.IsDir() {
+				return nil
+			}
```
Same comment here I guess :D

r-vasquez replied:
Fixed in https://github.com/redpanda-data/redpanda/compare/0404d15560b25c7026b593382887829c77a03ceb..044debc20df7bbda0e0bf68653119370a0f08b4a 

Thanks.

r-vasquez commented:
/ci-repeat 5
skip-units
dt-repeat=10
tests/rptest/tests/rpk_registry_test.py::RpkRegistryTest.test_produce_consume_proto

vbotbuildovich commented:
<!--NEW FAILURES-->
<!--714e1f4d-c8d1-439d-8fff-c2947e292fe9-->
<!--NEW FAILURES-->
<!--25164eb1-ceb5-405c-a526-813fd0aa2d89-->
<!--NEW FAILURES-->
<!--a0c15996-59f3-4ff8-a79a-6e31b19ce9f9-->
<!--NEW FAILURES-->
<!--d964d724-1ae1-4cf4-ac84-1ba2fddad3bb-->
<!--NEW FAILURES-->
<!--74ef21f3-1ce7-4797-ac23-0c5728d4630f-->
<!--NEW FAILURES-->
<!--b67e0cc0-a05f-4a5e-944a-98eba980f3d4-->
<!--NEW FAILURES-->
<!--9e032a48-4c63-48ad-9cbb-bbb9cd824ec9-->
<!--NEW FAILURES-->
<!--7cc5691c-23aa-4053-a030-a2b416b1dea0-->
<!--NEW FAILURES-->
<!--224e78dc-4364-49c5-a4cb-939b61d96bb0-->
<!--NEW FAILURES-->
<!--059d3c7a-a422-4d1f-8b41-4b0ff8aeba09-->
<!--NEW FAILURES-->
<!--c9dd29c0-8152-43d1-a153-d07ccb4b0900-->
<!--NEW FAILURES-->
<!--ab0fe38a-807b-467e-97e1-3e1f1c5cca80-->
<!--NEW FAILURES-->
<!--dda2cddf-8630-493f-8d1a-fd1501a67b26-->
<!--NEW FAILURES-->
<!--7dbbf951-d58b-4cc3-b3f2-780926152990-->
<!--NEW FAILURES-->
<!--7e5d54f9-52e4-471c-9191-b922f8c8112a-->
<!--NEW FAILURES-->
<!--1704b9ec-1789-4c7b-a97c-25ad3586e34b-->
<!--body-->
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59579#0193b3ae-60fa-4734-ba97-98c7212fb0a4:
```
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59579#0193b3ae-60fb-470b-9acc-b1534cabb534:
```
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59579#0193b3b2-aa79-4c97-8539-0359a23f9c60:
```
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59579#0193b3b2-aa79-40cc-83ef-b76850f4d4ea:
```
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59579#0193b3b2-aa78-47f7-a35d-5ff02a6f8fee:
```
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59579#0193b3b2-aa79-4ff0-a0ac-290444bd534d:
```
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59639#0193b9bd-8d23-4304-837c-d100a6b95cf8:
```
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59639#0193b9bd-8d22-4828-82c2-d53ee6ac7a9a:
```
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59639#0193b9bd-8d24-46bd-81b2-68272fd075fe:
```
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59639#0193b9bd-8d23-4e4e-9860-f76ef437691c:
```
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59639#0193b9bd-8d24-4817-8e85-24a1e5049d07:
```
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59639#0193b9c2-0138-42c3-b238-2c91422e373a:
```
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59639#0193b9c2-0139-48cc-b8f4-cdeedba5a942:
```
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59639#0193b9c2-0138-4198-8b6d-34799f487dd1:
```
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59639#0193b9c2-0139-44cb-b8df-4bb408d803e9:
```
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59639#0193b9c2-0139-455b-9d31-b3fee2811850:
```
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
"rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto"
```
<!--/body-->

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/59579-->
<!--f9fb966c-e753-4335-a774-5c85c7f30832-->
<!--db4b9983-a8de-443c-bd37-396666257f4b-->
<!--f209baec-9fa7-4530-bfd8-5cc56a519887-->
<!--c31b6367-39e4-4509-b1c5-4ee4319fff47-->
<!--3e1ddcb0-57f6-4d9f-9cbc-61f0d141d629-->
<!--a0952050-80d4-4e49-a664-a3ff20389c3a-->
<!--title-->
## Retry command for [Build#59579](https://buildkite.com/redpanda/redpanda/builds/59579)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/rpk_registry_test.py::RpkRegistryTest.test_produce_consume_proto
```
<!--/body-->

r-vasquez commented:
/ci-repeat 5
skip-units
dt-repeat=10
tests/rptest/tests/rpk_registry_test.py::RpkRegistryTest.test_produce_consume_proto

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/59639-->
<!--4c03b3b8-9cd0-42c1-8cf4-d5f2a9a83872-->
<!--8a6da1f4-b7b5-46eb-a261-4200e11d5d62-->
<!--69a65f3b-6dcb-48d6-b3fe-49a17d327f27-->
<!--c46d75bf-be8d-4082-a404-3587defcf304-->
<!--ec97c4ee-f516-412d-8b3a-127c13957769-->
<!--764b3379-9aef-4373-a7e8-9444bc90dadf-->
<!--12717ba6-b8ce-4083-a92a-35ae7db41e5d-->
<!--50b011d4-4b98-42a8-9ba8-d6c9bec5d3a9-->
<!--bee1f5ef-24a8-4a43-81a1-9c0c575c025d-->
<!--60e0545a-0eb8-4d2b-a996-ff4186bc21ec-->
<!--title-->
## Retry command for [Build#59639](https://buildkite.com/redpanda/redpanda/builds/59639)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/rpk_registry_test.py::RpkRegistryTest.test_produce_consume_proto
```
<!--/body-->

r-vasquez commented:
/ci-repeat 5
skip-units
dt-repeat=10
tests/rptest/tests/rpk_registry_test.py::RpkRegistryTest.test_produce_consume_proto

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/59669-->
<!--652bf620-5830-4a90-93ca-28ce2f7cbb1a-->
<!--19c9babd-344e-43f9-ac12-13876a765104-->
<!--bdd12e00-2531-48d1-ae63-8a068bf06829-->
<!--title-->
## Retry command for [Build#59669](https://buildkite.com/redpanda/redpanda/builds/59669)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```


/ci-repeat 1
tests/rptest/tests/rpk_registry_test.py::RpkRegistryTest.test_produce_consume_proto
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--948bc1cf-f031-4528-b840-31d0d478b301-->
<!--TEST RESULTS-->
<!--01aa690a-25e6-4074-a761-58c4229e0968-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/59669">build#59669</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:--------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto | ducktape | https://buildkite.com/redpanda/redpanda/builds/59669#0193bc10-c363-4098-add2-b33de66d103a | FAIL | 0/10 |
| rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto | ducktape | https://buildkite.com/redpanda/redpanda/builds/59669#0193bc10-c363-4afe-b095-9a9e11cbcff3 | FLAKY | 4/10 |
| rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto | ducktape | https://buildkite.com/redpanda/redpanda/builds/59669#0193bc22-f52d-41a9-b79b-a18d106db098 | FLAKY | 5/10 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/59677">build#59677</a></summary>


| test_id                                                                   | test_kind   | job_url                                                                                   | test_status   | passed   |
|:--------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto | ducktape    | https://buildkite.com/redpanda/redpanda/builds/59677#0193bcb0-359b-406e-887c-67727c480181 | FLAKY         | 4/10     |
| rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto | ducktape    | https://buildkite.com/redpanda/redpanda/builds/59677#0193bcb0-359b-4414-bafc-27dbbad8af8b | FLAKY         | 2/10     |
| rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto | ducktape    | https://buildkite.com/redpanda/redpanda/builds/59677#0193bcb0-359b-4683-a9ff-09871cbdadf2 | FLAKY         | 4/10     |
| rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto | ducktape    | https://buildkite.com/redpanda/redpanda/builds/59677#0193bcb0-359c-4464-a9ac-b9293f5b4175 | FLAKY         | 4/10     |
| rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto | ducktape    | https://buildkite.com/redpanda/redpanda/builds/59677#0193bcb0-359c-44af-904c-5d425295a53d | FLAKY         | 3/10     |
| rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto | ducktape    | https://buildkite.com/redpanda/redpanda/builds/59677#0193bcbd-6fea-47a5-9cc6-8b7c38dbf370 | FLAKY         | 5/10     |
| rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto | ducktape    | https://buildkite.com/redpanda/redpanda/builds/59677#0193bcbd-6fea-4a83-a7f1-3e35835f167f | FLAKY         | 6/10     |
| rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto | ducktape    | https://buildkite.com/redpanda/redpanda/builds/59677#0193bcbd-6feb-484f-961d-ac3f943e2129 | FLAKY         | 4/10     |
| rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto | ducktape    | https://buildkite.com/redpanda/redpanda/builds/59677#0193bcbd-6fec-43d1-8105-f25685ffff8f | FLAKY         | 4/10     |
| rptest.tests.rpk_registry_test.RpkRegistryTest.test_produce_consume_proto | ducktape    | https://buildkite.com/redpanda/redpanda/builds/59677#0193bcbd-6fec-4801-a9b2-0992e99213da | FLAKY         | 4/10     |
</details>
<!--/body-->

r-vasquez commented:
/ci-repeat 5
skip-units
dt-repeat=10
tests/rptest/tests/rpk_registry_test.py::RpkRegistryTest.test_produce_consume_proto

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/59677-->
<!--9df2d836-942f-482b-8bb4-23cbc3bf3e9d-->
<!--0c4a0008-a96e-4ebf-9445-fa9d7aa6bd0f-->
<!--8834ea5a-1f60-4657-bf4d-2599e3c8442d-->
<!--396b7a96-6c87-49f2-8b51-e56c6cf812c9-->
<!--11aeda04-af9e-4cdf-99eb-72acbf50fdc8-->
<!--7ac0f17b-4776-4bd4-b0f5-550f79b14bd0-->
<!--8a4706ab-9bb5-4d68-8040-be4997da4dbf-->
<!--bfcb32c0-4c52-41e6-8953-37985cff3054-->
<!--76b66dc0-6681-4619-84ed-bc26fde117b2-->
<!--49b7b53c-1654-4fc5-9fd6-7d6e4da0e8d4-->
<!--title-->
## Retry command for [Build#59677](https://buildkite.com/redpanda/redpanda/builds/59677)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```


/ci-repeat 1
tests/rptest/tests/rpk_registry_test.py::RpkRegistryTest.test_produce_consume_proto
```
<!--/body-->

r-vasquez commented:
/ci-repeat 2
skip-units
dt-repeat=5
tests/rptest/tests/rpk_registry_test.py::RpkRegistryTest.test_produce_consume_proto

r-vasquez commented:
Sorry for the noise with all the repeated tests @bojand, @rockwotj we ended up discovering an issue with parallel runs in our CI (see DEVPROD-2506).

Now the PR is ready for review again. The main changes since the last review is in the ducktape test (increasing timeouts, and hardening the SR startup)

r-vasquez commented:
/backport v24.3.x
